index.js

```diff
@@ -30,7 +35,16 @@ const streamFeaturesFromFile = (filePath) => {
       ) {
         return streamFromMapnik(featureStream, filePath, fileType);
       }
-      throw invalid(`Unknown file type "${fileType}": accepts .geojson, .csv, or .shp`);
+      if (fileType === 'zip') {
+        shpFairy(filePath, (err, shpFile) => {
+          if (err && err.code === 'EINVALID') {
+            return handleError(invalid('Invalid zipfile: ' + err.message));
+          }
+          if (err) return handleError(err);
+          return streamFromMapnik(featureStream, shpFile, 'shp');
+        });
+      }
+      else return handleError(invalid(`Unknown file type "${fileType}": accepts .geojson, .csv, or .shp (zipped and unzipped)`));
     })
     .catch((err) => featureStream.emit('error', err));
```

Comment:
Love it, thanks for the input @davidtheclark . I'll give your `streamFromZip` idea a whirl.